rasa/core/nlg/template.py

```diff
@@ -104,25 +104,33 @@ def _fill_template_text(
         template_vars = self._template_variables(filled_slots, kwargs)
 
         # Filling the template variables in the template text
-        if template_vars and "text" in template:
-            try:
-                # transforming template tags from
-                # "{tag_name}" to "{0[tag_name]}"
-                # as described here:
-                # https://stackoverflow.com/questions/7934620/python-dots-in-the-name-of-variable-in-a-format-string#comment9695339_7934969
-                # assuming that slot_name do not contain newline character here
-                text = re.sub(r"{([^\n]+?)}", r"{0[\1]}", template["text"])
-                template["text"] = text.format(template_vars)
-            except KeyError as e:
-                logger.exception(
-                    "Failed to fill utterance template '{}'. "
-                    "Tried to replace '{}' but could not find "
-                    "a value for it. There is no slot with this "
-                    "name nor did you pass the value explicitly "
-                    "when calling the template. Return template "
-                    "without filling the template. "
-                    "".format(template, e.args[0])
-                )
+        if template_vars:
+            if "text" in template:
+                try:
+                    template["text"] = interpolate_text(template["text"], template_vars)
+                except KeyError as e:
+                    logger.exception(
+                        "Failed to fill utterance template '{}'. "
```

Comment:
As we are throwing the exact same exception for both template types, I would suggest to either move the try/except up one level or put the exception message into a constant.